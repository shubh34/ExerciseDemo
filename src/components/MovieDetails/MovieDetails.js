import React from 'react';

const MovieDetails = () => <div>this is MovieDetails page </div>;

export default MovieDetails;
