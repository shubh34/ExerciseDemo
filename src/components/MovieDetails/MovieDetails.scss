.movie-details {
    @media screen and (min-width: 1140px) {
        padding-top: 60px;
    }
}

.movie-info {
    position: relative;
    height: 100%;
}
.details_actions {
    position: relative;
    top: 30%;
   
}
.details_meta {
    position: absolute;
    bottom: -16px;
    display: block;
    width: 100%
}
.details_meta_title {
    text-align: center;
}
.movie-details-description {
    padding: 16px;
    @media screen and (min-width: 1140px) {
        padding-top: 24px;
    }
    letter-spacing: 0.5px;
    line-height: 32px;
}
