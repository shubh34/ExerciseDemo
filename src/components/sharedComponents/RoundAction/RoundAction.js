import React from 'react';
import PropTypes from 'prop-types';
import './RoundAction.scss';

const RoundAction = (props) => {
	const { element, title } = props;
	return (
		<div className="round-action">
			<button type="button" className="round-action_element"/>
			<span className="round-action_title">{title}</span>
		</div>
	);
};

RoundAction.propTypes = {
	title: PropTypes.string.isRequired,
	element: PropTypes.node.isRequired,
};
export default RoundAction;
