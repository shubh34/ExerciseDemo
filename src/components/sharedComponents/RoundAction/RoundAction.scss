.round-action_element {
    width: 80px;
    height: 80px;
    border-radius: 50%;
    box-shadow: 0 1px 1px rgba(0,0,0,.5);
    background: transparent;
    &:before {
        content: '\e614';
        color: #fff;
        font-family: Glyphs;
        font-size: 40px;
        transition: color .2s ease;
    }
}
