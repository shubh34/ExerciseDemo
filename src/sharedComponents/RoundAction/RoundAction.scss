.round-action {
    display: flex;
    flex-direction: column;
    align-items: center;
}
.round-action_element {
    width: 80px;
    height: 80px;
    border-radius: 50%;
    box-shadow: 0 1px 1px rgba(0,0,0,.5);
    background-color: rgba(0,0,0,.7);
    border: none;
    cursor: pointer;
    &:before {
        content: '\e614';
        color: #fff;
        font-family: Glyphs;
        font-size: 40px;
        transition: color .2s ease;
       
    }
    &:hover {
        &:before {
            color: #ffe300;
        }
    }
    @media screen and (min-width: 1140px) {
        width: 110px;
        height: 110px;
        &:before {
            font-size: 56px;
        }
    }
}
.round-action_title {
    padding: 8px;
}
