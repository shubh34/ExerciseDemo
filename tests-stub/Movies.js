/* eslint-disable max-len */
export const rawMovies = {
	'populares-en-taquilla': {
		data: {
			type: 'lists',
			id: 'populares-en-taquilla',
			numerical_id: 4015,
			category: 'rent',
			short_name: 'Popular Alquiler',
			content_type: 'Movie',
			is_b2b: false,
			wktv_code: null,
			only_coupon: false,
			kind: 'regular',
			is_recommendation: false,
			name: 'Películas más vistas en Alquiler',
			additional_images: {},
			contents: {
				data: [{
					type: 'movies',
					id: 'capitana-marvel',
					numerical_id: 86665,
					title: 'Capitana Marvel',
					year: 2019,
					duration: 124,
					label: '3,99 €',
					classification: {
						type: 'classifications', id: '3', numerical_id: 3, name: '7', age: 7, adult: false, description: 'Mostrar sólo los contenidos autorizados para niños de hasta 7 años', default: false,
					},
					images: { artwork: 'https://images-2.wuaki.tv/system/artworks/86665/master/capitana-marvel-1560352921.jpeg', snapshot: 'https://images-3.wuaki.tv/system/shots/144189/original/capitana-marvel-1560352960.jpeg', ribbons: [] },
					highlighted_score: { score: 7.0, amount_of_votes: 6482, formatted_amount_of_votes: '6,5K' },
					rating: { average: 4, scale: 5 },
					labels: {
						audio_qualities: [{
							type: 'audio_qualities', id: '2.0', numerical_id: 2, name: '2.0 (Stereo)', abbr: '2.0', image: 'https://images-1.wuaki.tv/system/logos/2/original/2-0-stereo-1456155689.png',
						}],
						hdr_types: [{
							type: 'hdr_types', id: 'NONE', numerical_id: 1, name: 'SDR', abbr: 'NONE', image: null,
						}],
						purchase_types: [{
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}],
						video_qualities: [{
							type: 'video_qualities', id: 'SD', numerical_id: 2, name: 'SD', abbr: 'SD', position: 0, image: 'https://images-1.wuaki.tv/system/logos/2/original/sd-1456155688.png',
						}, {
							type: 'video_qualities', id: 'HD', numerical_id: 1, name: 'HD', abbr: 'HD', position: 1, image: 'https://images-0.wuaki.tv/system/logos/1/original/hd-1456155687.png',
						}],
					},
				}, {
					type: 'movies',
					id: 'a-47-metros',
					numerical_id: 54649,
					title: 'A 47 metros',
					year: 2017,
					duration: 89,
					label: '1,99 €',
					classification: {
						type: 'classifications', id: '7', numerical_id: 7, name: '12', age: 12, adult: false, description: 'Mostrar sólo los contenidos autorizados para los niños de hasta 12 años', default: false,
					},
					images: { artwork: 'https://images-3.wuaki.tv/system/artworks/54649/master/a-47-metros-1558460397.jpeg', snapshot: 'https://images-3.wuaki.tv/system/shots/84261/original/a-47-metros-1512392615.jpeg', ribbons: [] },
					highlighted_score: { score: 5.6, amount_of_votes: 1553, formatted_amount_of_votes: '1,6K' },
					rating: { average: 3, scale: 5 },
					labels: {
						audio_qualities: [{
							type: 'audio_qualities', id: '2.0', numerical_id: 2, name: '2.0 (Stereo)', abbr: '2.0', image: 'https://images-1.wuaki.tv/system/logos/2/original/2-0-stereo-1456155689.png',
						}],
						hdr_types: [{
							type: 'hdr_types', id: 'NONE', numerical_id: 1, name: 'SDR', abbr: 'NONE', image: null,
						}],
						purchase_types: [{
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}],
						video_qualities: [{
							type: 'video_qualities', id: 'SD', numerical_id: 2, name: 'SD', abbr: 'SD', position: 0, image: 'https://images-1.wuaki.tv/system/logos/2/original/sd-1456155688.png',
						}, {
							type: 'video_qualities', id: 'HD', numerical_id: 1, name: 'HD', abbr: 'HD', position: 1, image: 'https://images-0.wuaki.tv/system/logos/1/original/hd-1456155687.png',
						}],
					},
				}, {
					type: 'movies',
					id: 'green-book',
					numerical_id: 86659,
					title: 'Green Book',
					year: 2019,
					duration: 129,
					label: '3,99 €',
					classification: {
						type: 'classifications', id: '7', numerical_id: 7, name: '12', age: 12, adult: false, description: 'Mostrar sólo los contenidos autorizados para los niños de hasta 12 años', default: false,
					},
					images: { artwork: 'https://images-2.wuaki.tv/system/artworks/86659/master/green-book-1558483188.jpeg', snapshot: 'https://images-2.wuaki.tv/system/shots/140827/original/green-book-1557828040.jpeg', ribbons: [] },
					highlighted_score: { score: 8.3, amount_of_votes: 3986, formatted_amount_of_votes: '4K' },
					rating: { average: 5, scale: 5 },
					labels: {
						audio_qualities: [{
							type: 'audio_qualities', id: '2.0', numerical_id: 2, name: '2.0 (Stereo)', abbr: '2.0', image: 'https://images-1.wuaki.tv/system/logos/2/original/2-0-stereo-1456155689.png',
						}],
						hdr_types: [{
							type: 'hdr_types', id: 'NONE', numerical_id: 1, name: 'SDR', abbr: 'NONE', image: null,
						}],
						purchase_types: [{
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}],
						video_qualities: [{
							type: 'video_qualities', id: 'HD', numerical_id: 1, name: 'HD', abbr: 'HD', position: 1, image: 'https://images-0.wuaki.tv/system/logos/1/original/hd-1456155687.png',
						}, {
							type: 'video_qualities', id: 'SD', numerical_id: 2, name: 'SD', abbr: 'SD', position: 0, image: 'https://images-1.wuaki.tv/system/logos/2/original/sd-1456155688.png',
						}],
					},
				}, {
					type: 'movies',
					id: 'mascotas',
					numerical_id: 43779,
					title: 'Mascotas',
					year: 2016,
					duration: 86,
					label: '2,99 €',
					classification: {
						type: 'classifications', id: '1', numerical_id: 1, name: 'T', age: 0, adult: false, description: 'Mostrar sólo contenidos autorizados para todos los públicos', default: false,
					},
					images: {
						artwork: 'https://images-2.wuaki.tv/system/artworks/43779/master/mascotas-1558452275.jpeg',
						snapshot: 'https://images-2.wuaki.tv/system/shots/70669/original/mascotas-1559026402.jpeg',
						ribbons: [{
							type: 'ribbons', id: '6', numerical_id: 6, name: 'UHD', color: '#ffcc00', text_color: '#000000', position: 'top-left', localized_name: 'UHD', image: 'https://images-1.wuaki.tv/system/images/6/original/uhd-1418228576.png',
						}],
					},
					highlighted_score: { score: 6.1, amount_of_votes: 5555, formatted_amount_of_votes: '5,6K' },
					rating: { average: 4, scale: 5 },
					labels: {
						audio_qualities: [{
							type: 'audio_qualities', id: '2.0', numerical_id: 2, name: '2.0 (Stereo)', abbr: '2.0', image: 'https://images-1.wuaki.tv/system/logos/2/original/2-0-stereo-1456155689.png',
						}],
						hdr_types: [{
							type: 'hdr_types', id: 'NONE', numerical_id: 1, name: 'SDR', abbr: 'NONE', image: null,
						}],
						purchase_types: [{
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}],
						video_qualities: [{
							type: 'video_qualities', id: 'HD', numerical_id: 1, name: 'HD', abbr: 'HD', position: 1, image: 'https://images-0.wuaki.tv/system/logos/1/original/hd-1456155687.png',
						}, {
							type: 'video_qualities', id: 'SD', numerical_id: 2, name: 'SD', abbr: 'SD', position: 0, image: 'https://images-1.wuaki.tv/system/logos/2/original/sd-1456155688.png',
						}],
					},
				}, {
					type: 'movies',
					id: 'mula',
					numerical_id: 89878,
					title: 'Mula',
					year: 2018,
					duration: 116,
					label: '4,99 €',
					classification: {
						type: 'classifications', id: '7', numerical_id: 7, name: '12', age: 12, adult: false, description: 'Mostrar sólo los contenidos autorizados para los niños de hasta 12 años', default: false,
					},
					images: {
						artwork: 'https://images-2.wuaki.tv/system/artworks/89878/master/mula-1558485497.jpeg',
						snapshot: 'https://images-0.wuaki.tv/system/shots/139695/original/mula-1557230190.jpeg',
						ribbons: [{
							type: 'ribbons', id: '6', numerical_id: 6, name: 'UHD', color: '#ffcc00', text_color: '#000000', position: 'top-left', localized_name: 'UHD', image: 'https://images-1.wuaki.tv/system/images/6/original/uhd-1418228576.png',
						}],
					},
					highlighted_score: { score: 6.5, amount_of_votes: 1769, formatted_amount_of_votes: '1,8K' },
					rating: { average: 4, scale: 5 },
					labels: {
						audio_qualities: [{
							type: 'audio_qualities', id: '2.0', numerical_id: 2, name: '2.0 (Stereo)', abbr: '2.0', image: 'https://images-1.wuaki.tv/system/logos/2/original/2-0-stereo-1456155689.png',
						}],
						hdr_types: [{
							type: 'hdr_types', id: 'NONE', numerical_id: 1, name: 'SDR', abbr: 'NONE', image: null,
						}],
						purchase_types: [{
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}],
						video_qualities: [{
							type: 'video_qualities', id: 'SD', numerical_id: 2, name: 'SD', abbr: 'SD', position: 0, image: 'https://images-1.wuaki.tv/system/logos/2/original/sd-1456155688.png',
						}, {
							type: 'video_qualities', id: 'HD', numerical_id: 1, name: 'HD', abbr: 'HD', position: 1, image: 'https://images-0.wuaki.tv/system/logos/1/original/hd-1456155687.png',
						}],
					},
				}, {
					type: 'movies',
					id: 'escape-room-2019',
					numerical_id: 89879,
					title: 'Escape Room (2019)',
					year: 2019,
					duration: 99,
					label: '3,99 €',
					classification: {
						type: 'classifications', id: '7', numerical_id: 7, name: '12', age: 12, adult: false, description: 'Mostrar sólo los contenidos autorizados para los niños de hasta 12 años', default: false,
					},
					images: {
						artwork: 'https://images-0.wuaki.tv/system/artworks/89879/master/escape-room-2019-1559899512.jpeg',
						snapshot: 'https://images-0.wuaki.tv/system/shots/143853/original/escape-room-2019-1559899363.jpeg',
						ribbons: [{
							type: 'ribbons', id: '6', numerical_id: 6, name: 'UHD', color: '#ffcc00', text_color: '#000000', position: 'top-left', localized_name: 'UHD', image: 'https://images-1.wuaki.tv/system/images/6/original/uhd-1418228576.png',
						}],
					},
					highlighted_score: { score: 6.4, amount_of_votes: 1523, formatted_amount_of_votes: '1,5K' },
					rating: { average: 4, scale: 5 },
					labels: {
						audio_qualities: [{
							type: 'audio_qualities', id: '2.0', numerical_id: 2, name: '2.0 (Stereo)', abbr: '2.0', image: 'https://images-1.wuaki.tv/system/logos/2/original/2-0-stereo-1456155689.png',
						}],
						hdr_types: [{
							type: 'hdr_types', id: 'NONE', numerical_id: 1, name: 'SDR', abbr: 'NONE', image: null,
						}],
						purchase_types: [{
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}],
						video_qualities: [{
							type: 'video_qualities', id: 'HD', numerical_id: 1, name: 'HD', abbr: 'HD', position: 1, image: 'https://images-0.wuaki.tv/system/logos/1/original/hd-1456155687.png',
						}, {
							type: 'video_qualities', id: 'SD', numerical_id: 2, name: 'SD', abbr: 'SD', position: 0, image: 'https://images-1.wuaki.tv/system/logos/2/original/sd-1456155688.png',
						}],
					},
				}, {
					type: 'movies',
					id: 'perdiendo-el-este',
					numerical_id: 86635,
					title: 'Perdiendo el Este',
					year: 2019,
					duration: 85,
					label: '4,99 €',
					classification: {
						type: 'classifications', id: '3', numerical_id: 3, name: '7', age: 7, adult: false, description: 'Mostrar sólo los contenidos autorizados para niños de hasta 7 años', default: false,
					},
					images: { artwork: 'https://images-1.wuaki.tv/system/artworks/86635/master/perdiendo-el-este-1559228752.jpeg', snapshot: 'https://images-3.wuaki.tv/system/shots/142994/original/perdiendo-el-este-1559228799.jpeg', ribbons: [] },
					highlighted_score: { score: 4.2, amount_of_votes: 20, formatted_amount_of_votes: '20' },
					rating: { average: 3, scale: 5 },
					labels: {
						audio_qualities: [{
							type: 'audio_qualities', id: '2.0', numerical_id: 2, name: '2.0 (Stereo)', abbr: '2.0', image: 'https://images-1.wuaki.tv/system/logos/2/original/2-0-stereo-1456155689.png',
						}],
						hdr_types: [{
							type: 'hdr_types', id: 'NONE', numerical_id: 1, name: 'SDR', abbr: 'NONE', image: null,
						}],
						purchase_types: [{
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}],
						video_qualities: [{
							type: 'video_qualities', id: 'SD', numerical_id: 2, name: 'SD', abbr: 'SD', position: 0, image: 'https://images-1.wuaki.tv/system/logos/2/original/sd-1456155688.png',
						}, {
							type: 'video_qualities', id: 'HD', numerical_id: 1, name: 'HD', abbr: 'HD', position: 1, image: 'https://images-0.wuaki.tv/system/logos/1/original/hd-1456155687.png',
						}],
					},
				}, {
					type: 'movies',
					id: 'creed-ii-la-leyenda-de-rocky',
					numerical_id: 79289,
					title: 'Creed II: la leyenda de rocky',
					year: 2018,
					duration: 129,
					label: '4,99 €',
					classification: {
						type: 'classifications', id: '8', numerical_id: 8, name: '16', age: 16, adult: false, description: 'Mostrar sólo los contenidos autorizados para los niños de hasta 16 años', default: false,
					},
					images: {
						artwork: 'https://images-3.wuaki.tv/system/artworks/79289/master/creed-ii-la-leyenda-de-rocky-1558478964.jpeg',
						snapshot: 'https://images-3.wuaki.tv/system/shots/144830/original/creed-ii-la-leyenda-de-rocky-1560523036.jpeg',
						ribbons: [{
							type: 'ribbons', id: '6', numerical_id: 6, name: 'UHD', color: '#ffcc00', text_color: '#000000', position: 'top-left', localized_name: 'UHD', image: 'https://images-1.wuaki.tv/system/images/6/original/uhd-1418228576.png',
						}],
					},
					highlighted_score: { score: 6.7, amount_of_votes: 2334, formatted_amount_of_votes: '2,3K' },
					rating: { average: 4, scale: 5 },
					labels: {
						audio_qualities: [{
							type: 'audio_qualities', id: '2.0', numerical_id: 2, name: '2.0 (Stereo)', abbr: '2.0', image: 'https://images-1.wuaki.tv/system/logos/2/original/2-0-stereo-1456155689.png',
						}],
						hdr_types: [{
							type: 'hdr_types', id: 'NONE', numerical_id: 1, name: 'SDR', abbr: 'NONE', image: null,
						}],
						purchase_types: [{
							type: 'purchase_types', id: '1', numerical_id: 1, is_recurring: false, name: 'Alquiler 48H', label: 'ALQUILER 48H', kind: 'rental', expires_after_in_seconds: 172800, available_time_in_seconds: 172800,
						}, {
							type: 'purchase_types', id: '2', numerical_id: 2, is_recurring: false, name: 'Venta (Digital Locker)', label: 'VENTA (DIGITAL LOCKER)', kind: 'purchase', expires_after_in_seconds: null, available_time_in_seconds: null,
						}],
						video_qualities: [{
							type: 'video_qualities', id: 'SD', numerical_id: 2, name: 'SD', abbr: 'SD', position: 0, image: 'https://images-1.wuaki.tv/system/logos/2/original/sd-1456155688.png',
						}, {
							type: 'video_qualities', id: 'HD', numerical_id: 1, name: 'HD', abbr: 'HD', position: 1, image: 'https://images-0.wuaki.tv/system/logos/1/original/hd-1456155687.png',
						}],
					},
				}],
				meta: {
					pagination: {
						page: 1, count: 49, per_page: 18, offset: 0, total_pages: 3,
					},
				},
			},
		},
	},
};
