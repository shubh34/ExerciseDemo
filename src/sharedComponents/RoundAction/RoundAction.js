import React from 'react';
import PropTypes from 'prop-types';
import './RoundAction.scss';

const RoundAction = (props) => {
	const { title = 'Trailer', onClick } = props;
	return (
		<div className="round-action">
			<button type="button" className="round-action_element" onClick={onClick} />
			<span className="round-action_title">{title}</span>
		</div>
	);
};

RoundAction.propTypes = {
	title: PropTypes.string,
	onClick: PropTypes.func.isRequired,
};
export default RoundAction;
