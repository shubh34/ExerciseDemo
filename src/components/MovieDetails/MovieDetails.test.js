import React from 'react';
import { shallow } from 'enzyme';

import { MovieDetails } from './MovieDetails';
import { movieDetailsStub } from '../../../tests-stub/MovieDetails';
import {
	getSnapshotImageUrl,
	getMovieTitle,
	getMovieDescription,
	getScore,
	getVotes
} from '../../states/movieDetails/selectors';
import RoundAction from '../../sharedComponents/RoundAction/RoundAction';

const setUp = (setUpProps = {}, container = shallow) => {
	const state = { movieDetails: movieDetailsStub };
	const props = {
		snapShotUrl: getSnapshotImageUrl(state),
		movieTitle: getMovieTitle(state),
		movieDescription: getMovieDescription(state),
		score: getScore(state),
		votes: getVotes(state),
		getMovieDetails: jest.fn(),
		resetMovieDetails: jest.fn(),
		onShowTrailor: jest.fn(),
		match: {
			params: {
				id: 'test'
			}
		},
		history: {
			push: jest.fn()
		},
		...setUpProps
	};
	const wrapper = container(<MovieDetails {...props} />);
	return { wrapper, props };
};

describe('MovieDetails', () => {
	it('should render details of movie', () => {
		const { wrapper } = setUp();
		expect(wrapper).toMatchSnapshot();
	});
	it('should navigate to trailor page', () => {
		const { wrapper, props } = setUp();
		wrapper.find(RoundAction).simulate('click');
		expect(props.history.push).toBeCalledWith('/streams/movie/test');
	});
});
