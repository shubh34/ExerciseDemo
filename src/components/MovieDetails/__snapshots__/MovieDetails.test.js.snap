// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails should render details of movie 1`] = `
<Fragment>
  <PageHeader />
  <div
    className="movie-details"
  >
    <Memo(MovieDetailsHero)
      heroUrl="https://images-3.wuaki.tv/system/shots/144189/original/capitana-marvel-1560352960-width1920-quality80.jpeg"
      onClick={[Function]}
    >
      <div
        className="movie-info"
      >
        <div
          className="details_actions"
        >
          <RoundAction
            onClick={[Function]}
          />
        </div>
        <div
          className="details_meta"
        >
          <Memo(Scores)
            className="movie-details-score"
            score={7}
            votes="6,5K"
          />
          <Header
            className="details_meta_title"
            header="Capitana Marvel"
          />
        </div>
      </div>
    </Memo(MovieDetailsHero)>
    <div
      className="movie-details-description"
    >
      <SectionHeader
        header="Descripción :"
      />
      <p>
        Capitana Marvel te lleva a una espectacular aventura en los años 90, siguiendo el camino de Carol Danvers (Brie Larson) mientras se convierte en una de las heroínas más poderosas del universo. Cuando la guerra galáctica llega a la Tierra, se encuentra con el agente Nick Furia (Samuel L. Jackson) en el centro del caos, que le lleva a su destino final como Vengador.
      </p>
    </div>
    <Connect(RecomendedComponent)
      movieId="test"
    />
  </div>
</Fragment>
`;
